src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/CodeActionEndToEndTest.cs

```diff
@@ -763,4 +828,45 @@ private void DoesNotExist()
 
         return changes;
     }
+
+    private class GenerateMethodResolverDocumentContextFactory : TestDocumentContextFactory
+    {
+        private readonly List<TagHelperDescriptor> _tagHelperDescriptors;
+
+        public GenerateMethodResolverDocumentContextFactory
+            (string filePath,
+            RazorCodeDocument codeDocument,
+            TagHelperDescriptor[]? tagHelpers = null,
+            int? version = null)
+            : base(filePath, codeDocument, version)
+        {
+
+            _tagHelperDescriptors = new() { CreateOnClickTagHelperDescriptor() };
```

Comment:
would love to see multiple tag helpers just to verify search works correctly in product code. (Looking at the code I know it does, but could easily change to `.First()` and these test would still work :) )